### PR TITLE
[ci] allow Dockerfile to come from inputs

### PR DESCRIPTION
**Pull request by danking**

### CODE REVIEW

**Review by daniel-goldstein — APPROVED** [review]
How did this work before…

**Comment by danking** [comment]
The docker file always came directly from the repository rather than from the inputs.

File: ci/test/resources/build.yaml [hunk 1]
```diff
@@ -55,6 +55,8 @@ steps:
     inputs:
       - from: /repo/docker/hail-ubuntu
         to: /inputs/docker/hail-ubuntu
+    dependsOn:
+      - code
```
Comment (daniel-goldstein) [inline ci/test/resources/build.yaml:59]: Isn't this from the other PR?

**Review by daniel-goldstein — CHANGES_REQUESTED** [review]
Some dockerfiles (e.g. Grafana) depend on a templated input file to render the Dockerfile. How could we have been rendering the Dockerfile before bringing in those inputs?

File: ci/test/resources/build.yaml (hunk 1 above)
Comment (danking) [reply ci/test/resources/build.yaml]: oh yep, that should be copy_files

**Comment by danking** [comment]
Grafana's Dockerfile is templated, but the template, `grafana/Dockerfile.nginx` came from the repository.

**Comment by daniel-goldstein** [comment]
Ah right, the dockerfile isn't actually used until after the inputs are brought in.

**Review by danking — COMMENTED** [review]
FYI Daniel, I got super frustrated trying to make this backwards compatible, so I just added a buildImage2 step, copied buildImage and removed all the unnecessary input manipulation and repository cloning.

After this PR and the next one go in, I can remove buildImage and rename buildImage2 to buildImage again.

**Comment by danking** [comment]
@daniel-goldstein ok fixed!

**Review by daniel-goldstein — APPROVED** [review]
LGTM

**Pull request merged**